DatadogSessionReplay/Sources/DatadogSessionReplay/DatadogCoreIntegration/RUMContextReceiver.swift

```diff
@@ -83,14 +85,20 @@ private extension DatadogContext {
 
 private extension FeatureBaggage {
     var rumContext: RUMContext? {
-        guard let applicationID: String = self[RUMDependency.applicationIDKey],
-              let sessionID: String = self[RUMDependency.sessionIDKey],
-              let viewID: String = self[RUMDependency.viewIDKey]
+        guard let applicationID: String = self[RUMDependency.ids, type: [String: String].self]?[RUMDependency.IDs.applicationIDKey],
+              let sessionID: String = self[RUMDependency.ids, type: [String: String].self]?[RUMDependency.IDs.sessionIDKey],
+              let viewID: String = self[RUMDependency.ids, type: [String: String].self]?[RUMDependency.IDs.viewIDKey],
+              let serverTimeOffset: TimeInterval = self[RUMDependency.serverTimeOffsetKey]
         else {
             // Current RUM session is not sampled
             return nil
         }
 
-        return RUMContext(applicationID: applicationID, sessionID: sessionID, viewID: viewID)
+        return RUMContext(
+            applicationID: applicationID,
+            sessionID: sessionID,
+            viewID: viewID,
+            serverTimeOffset: serverTimeOffset
+        )
```

Comment:
This is no longer necessary with the `AnyCoder`, you can make the `RUMContext` Decodable and let the `FeatureBaggage` do the decoding:

```swift
/// The RUM context received from `DatadogCore`.
internal struct RUMContext: Decodable, Equatable {
    enum CodingKeys: String, CodingKey {
        case applicationID = "application_id"
        case sessionID = "session_id"
        case viewID = "view.id"
    }

    /// Current RUM application ID - standard UUID string, lowecased.
    let applicationID: String
    /// Current RUM session ID - standard UUID string, lowecased.
    let sessionID: String
    /// Current RUM view ID - standard UUID string, lowecased.
    let viewID: String
}

private extension FeatureBaggage {
    var rumContext: RUMContext? { self[RUMDependency.ids] }

    /// Current RUM view server time offset (in seconds).
    var viewTimeOffset: TimeInterval? { self[RUMDependency.serverTimeOffsetKey] }
}
```

With this, the enum `RUMDependency.IDs` is no longer necessary.
I would also suggest to rename the `serverTimeOffset` to `viewTimeOffset` to make it clear that it's from the current session.

Reply:
The `FeatureBaggage` extension is actually not necessary, it already comes with `@dynamicMemberLookup` :)